rpcserver.go

```diff
@@ -1743,10 +1783,77 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 			"payment allowed is %v", amt, maxPaymentMSat.ToSatoshis())
 	}
 
+	// Next, generate the payment hash itself from the preimage. This will
+	// be used by clients to query for the state of a particular invoice.
+	rHash := sha256.Sum256(paymentPreimage[:])
+
+	// We also create an encoded payment request which allows the
+	// caller to compactly send the invoice to the payer. We'll create a
+	// list of options to be added to the encoded payment request. For now
+	// we only support the required fields description/description_hash,
+	// expiry, fallback address, and the amount field.
+	var options []func(*zpay32.Invoice)
+
+	// Add the amount. This field is optional by the BOLT-11 format, but
+	// we require it for now.
+	options = append(options, zpay32.Amount(amtMSat))
+
+	// If specified, add a fallback address to the payment request.
+	if len(invoice.FallbackAddr) > 0 {
+		addr, err := btcutil.DecodeAddress(invoice.FallbackAddr,
+			activeNetParams.Params)
+		if err != nil {
+			return nil, fmt.Errorf("invalid fallback address: %v",
+				err)
+		}
+		options = append(options, zpay32.FallbackAddr(addr))
+	}
+
+	// If expiry is set, specify it. If it is not provided, then expiry will
```

Comment:
I may be missing something, but, I don't see the location that the default expiration is enforced. Also note that the expiration should always be in _absolute_ terms. Not relative as it seems to be now.

Reply:
If it is not specified, it will not be encoded in the payment request. The receiver will know that a missing expiry means 3600 seconds. I updated the comment to make this more clear.

Expiration is set as `timestamp + expiry`, so expiry should in this case be relative.